pkg/controller/batchrelease/workloads/commons.go

```diff
@@ -243,3 +243,59 @@ func getPodOrdinal(pod *corev1.Pod) int {
 	ord, _ := strconv.Atoi(pod.Name[strings.LastIndex(pod.Name, "-")+1:])
 	return ord
 }
+
+func globalFailureThreshold(global, maxUnavailable *intstr.IntOrString, replicas int32) int32 {
+	globalThreshold := 0
+	if global != nil {
+		globalThreshold, _ = intstr.GetScaledValueFromIntOrPercent(global, int(replicas), true)
+	} else if maxUnavailable != nil {
+		globalThreshold, _ = intstr.GetScaledValueFromIntOrPercent(maxUnavailable, int(replicas), true)
+	}
+	return int32(integer.IntMax(0, globalThreshold))
+}
+
+func isBatchReady(release *v1alpha1.BatchRelease, pods []*corev1.Pod, maxUnavailable *intstr.IntOrString, planned, desired, updated, updatedReady int32) bool {
+	updateRevision := release.Status.UpdateRevision
+	updatedReady = integer.Int32Max(updatedReady, int32(util.WrappedPodCount(pods, func(pod *corev1.Pod) bool {
+		return pod.DeletionTimestamp.IsZero() && util.IsConsistentWithRevision(pod, updateRevision) && util.IsPodReady(pod)
+	})))
+
+	rolloutID := release.Spec.ReleasePlan.RolloutID
+	currentBatch := release.Status.CanaryStatus.CurrentBatch
+	globalThreshold := globalFailureThreshold(release.Spec.ReleasePlan.FailureThreshold, maxUnavailable, updated)
```

Comment:
I think there should be workload.spec.replicas, rather than the updated.

Reply:
It will block upgrade If all old Pods failed.